File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/util/UserGroupUtilTest.java
package no.nav.pensjon.selvbetjeningopptjening.util;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static no.nav.pensjon.selvbetjeningopptjening.model.code.UserGroup.*;
import static org.junit.jupiter.api.Assertions.assertEquals;

class UserGroupUtilTest {

    @Test
    void findUserGroup_1_when_birthYear_is_beforeNewAlder() {
        var beforeNewAlder = LocalDate.of(1942, 12, 31);
        assertEquals(USER_GROUP_1, UserGroupUtil.findUserGroup(beforeNewAlder));
    }

    @Test
    void findUserGroup_2_when_birthYear_is_beforeNewAfp() {
        var beforeNewAfp = LocalDate.of(1947, 12, 31);
        assertEquals(USER_GROUP_2, UserGroupUtil.findUserGroup(beforeNewAfp));
    }

    @Test
    void findUserGroup_3_when_birthYear_is_december1948() {
        var december1948 = LocalDate.of(1948, 12, 1);
        assertEquals(USER_GROUP_3, UserGroupUtil.findUserGroup(december1948));
    }

    @Test
    void findUserGroup_3_when_birthYear_is_beforeOvergangsregler() {
        var beforeOvergangsregler = LocalDate.of(1953, 12, 31);
        assertEquals(USER_GROUP_3, UserGroupUtil.findUserGroup(beforeOvergangsregler));
    }

    @Test
    void findUserGroup_4_when_birthYear_is_withinOvergangsregler() {
        var lastYearWithOvergangsregler = LocalDate.of(1962, 12, 31);
        assertEquals(USER_GROUP_4, UserGroupUtil.findUserGroup(lastYearWithOvergangsregler));
    }

    @Test
    void findUserGroup_5_when_birthYear_is_afterOvergangsregler() {
        var afterOvergangsregler = LocalDate.of(1963, 1, 31);
        assertEquals(USER_GROUP_5, UserGroupUtil.findUserGroup(afterOvergangsregler));
    }
}

File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/person/group/UserGroupUtilTest.kt
package no.nav.pensjon.selvbetjeningopptjening.person.group

import io.kotest.core.spec.style.ShouldSpec
import io.kotest.matchers.shouldBe
import no.nav.pensjon.selvbetjeningopptjening.model.code.UserGroup
import java.time.LocalDate

class UserGroupUtilTest : ShouldSpec({

    should("return user group 1 when birthYear is before new alder") {
        val beforeNewAlder = LocalDate.of(1942, 12, 31)
        UserGroupUtil.findUserGroup(beforeNewAlder) shouldBe UserGroup.USER_GROUP_1
    }

    should("return user group 2 when birthYear is before new privat AFP") {
        val beforeNewAfp = LocalDate.of(1947, 12, 31)
        UserGroupUtil.findUserGroup(beforeNewAfp) shouldBe UserGroup.USER_GROUP_2
    }

    should("return user group 3 when birthYear is December 1948") {
        val december1948 = LocalDate.of(1948, 12, 1)
        UserGroupUtil.findUserGroup(december1948) shouldBe UserGroup.USER_GROUP_3
    }

    should("return user group 3 when birthYear is before overgangsregler") {
        val beforeOvergangsregler = LocalDate.of(1953, 12, 31)
        UserGroupUtil.findUserGroup(beforeOvergangsregler) shouldBe UserGroup.USER_GROUP_3
    }

    should("return user group 4 when birthYear is within overgangsregler") {
        val lastYearWithOvergangsregler = LocalDate.of(1962, 12, 31)
        UserGroupUtil.findUserGroup(lastYearWithOvergangsregler) shouldBe UserGroup.USER_GROUP_4
    }

    should("return user group 5 when birthYear is after overgangsregler") {
        val afterOvergangsregler = LocalDate.of(1963, 1, 31)
        UserGroupUtil.findUserGroup(afterOvergangsregler) shouldBe UserGroup.USER_GROUP_5
    }
})
